hles/transformation/src/main/scala/org/broadinstitute/monster/dap/dog/DietTransformations.scala

```diff
@@ -43,9 +43,12 @@ object DietTransformations {
       dfEverUnderweight = rawRecord.getOptionalNumber("df_underweight"),
       // Note: "df_overrweight" raw field name is misspelled in RedCap
       dfEverOverweight = rawRecord.getOptionalNumber("df_overrweight"),
+      // "1" from gated question maps to "yes, weight stayed the same" so we inject 0
+      // "2" from gated question maps to "puppy weight gain" so we inject a 3 to keep the distinction
       dfWeightChangeLastYear = weightChange.flatMap {
         case 1L    => Some(0L)
         case 0L    => rawRecord.getOptionalNumber("df_weight_change_how")
```

Comment:
wouldn't `df_weight_change_how` be required if they answered `0L` here?

Reply:
Yup, that's not changing.
The options for the gated question are:
If we get a 1 we inject 0 into the follow up question's raw value
1 Yes
0 No
2 My dog is a puppy who was growing to his/her full size during the past year
99 Don't know

So we are correctly pulling the follow up when we hit 0. It's just that before this change we were passing through 2 as 2 for some reason. My thinking is that someone decided "My dog is a puppy who was growing to his/her full size during the past year" == "Gained Weight" in the follow up question - however DAP wants to keep the distinction by injecting "3" into the follow up response when we see a "2" in the gate.

Reply:
Sure, but I'm more referring to the use of `getOptionalNumber` rather than `getRequired` or similar for `df_weight_change_how` when the field is required if `df_weight_change` is answered with 0.

I don't think it's likely enough to fuss over too much, just something that jumped out at me.

Reply:
oh i see - yeah that's a good point.
actually we probably should enforce that, especially for these "gated question - follow up" scenarios. let me switch that to a getRequired and make sure everything compiles correctly